qpeerset/qpeerset.go

```diff
@@ -0,0 +1,164 @@
+package qpeerset
+
+import (
+	"math/big"
+	"sort"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	ks "github.com/whyrusleeping/go-keyspace"
+)
+
+type PeerState int
+
+const (
+	PeerSeen PeerState = iota
+	PeerWaiting
+	PeerQueried
+	PeerUnreachable
```

Comment:
I don't think we need this state at all. When a peer is unreachable, we can simply remove it. `GetClosestNotUnreachable` would then return  'seen' + 'queried' peers which would also serve the purpose.